code/game/objects/structures/catwalk.dm

```diff
@@ -1,107 +1,183 @@
-		if(health < maxhealth)
-			to_chat(user, "<span class='notice'>You begin repairing \the [src.name] with \the [C.name].</span>")
-			if(do_after(user, 20, src))
-				health = maxhealth
-	else
-		take_damage(C.force)
-		user.setClickCooldown(user.get_attack_speed(C))
-	return ..()
-
-/obj/structure/catwalk/Crossed()
-	. = ..()
-	if(isliving(usr) && !usr.is_incorporeal())
-		playsound(src, pick('sound/effects/footstep/catwalk1.ogg', 'sound/effects/footstep/catwalk2.ogg', 'sound/effects/footstep/catwalk3.ogg', 'sound/effects/footstep/catwalk4.ogg', 'sound/effects/footstep/catwalk5.ogg'), 25, 1)
+		if(WT.isOn() && WT.remove_fuel(0, user))
+			deconstruct(user)
+			return
+	if(C.is_crowbar() && plated_tile)
+		hatch_open = !hatch_open
+		if(hatch_open)
+			playsound(src, 'sound/items/Crowbar.ogg', 100, 2)
+			to_chat(user, "<span class='notice'>You pry open \the [src]'s maintenance hatch.</span>")
+		else
+			playsound(src, 'sound/items/Deconstruct.ogg', 100, 2)
+			to_chat(user, "<span class='notice'>You shut \the [src]'s maintenance hatch.</span>")
+		update_icon()
+		return
+	if(istype(C, /obj/item/stack/tile/floor) && !plated_tile)
+		var/obj/item/stack/tile/floor/ST = C
+		to_chat(user, "<span class='notice'>Placing tile...</span>")
+		if (!do_after(user, 10))
+			return
+		if(!ST.use(1))
+			return
+		to_chat(user, "<span class='notice'>You plate \the [src]</span>")
+		name = "plated catwalk"
+		plated_tile = C.type
+		src.add_fingerprint(user)
+		for(var/tiletype in plating_colors)
+			if(istype(ST, tiletype))
+				plating_color = plating_colors[tiletype]
+		update_icon()
 
-/obj/structure/catwalk/CheckExit(atom/movable/O, turf/target)
-	if(O.checkpass(PASSGRILLE))
-		return 1
-	if(target && target.z < src.z)
-		return 0
-	return 1
+/obj/structure/catwalk/refresh_neighbors()
+	return
 
 /obj/structure/catwalk/take_damage(amount)
 	health -= amount
 	if(health <= 0)
 		visible_message("<span class='warning'>\The [src] breaks down!</span>")
 		playsound(loc, 'sound/effects/grillehit.ogg', 50, 1)
 		new /obj/item/stack/rods(get_turf(src))
-		Destroy()
\ No newline at end of file
+		Destroy()
+
+/obj/structure/catwalk/Crossed()
+	. = ..()
+	if(isliving(usr) && !usr.is_incorporeal())
+		playsound(src, pick('sound/effects/footstep/catwalk1.ogg', 'sound/effects/footstep/catwalk2.ogg', 'sound/effects/footstep/catwalk3.ogg', 'sound/effects/footstep/catwalk4.ogg', 'sound/effects/footstep/catwalk5.ogg'), 25, 1)
+
+/obj/effect/catwalk_plated
+	name = "plated catwalk spawner"
+	icon = 'icons/turf/catwalks.dmi'
+	icon_state = "catwalk_plated"
+	density = 1
+	anchored = 1.0
+	var/activated = FALSE
+	plane = DECAL_PLANE
+	layer = ABOVE_UTILITY
+	var/tile = /obj/item/stack/tile/floor
+	var/platecolor = "#858a8f"
+
+/obj/effect/catwalk_plated/Initialize(mapload)
+	. = ..()
+	activate()
+
+/obj/effect/catwalk_plated/CanPass()
+	return 0
+
+/obj/effect/catwalk_plated/attack_hand()
+	attack_generic()
+
+/obj/effect/catwalk_plated/attack_ghost()
+	attack_generic()
```

Comment:
Does this do anything that shouldn't have already happened?

Reply:
Not really. Bay had some weird obsession with not having the plated catwalk spawners 'work' after the round had started. I removed most of that. For some reason they would rather ones spawned in-round require a click to 'finish spawning'. I left the click-makes-them-spawn because why not, but I removed the thing about them not spawning if they were created in-round (via admin spawn for eg).